base/compiler/abstractinterpretation.jl

```diff
@@ -979,8 +979,9 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter,
         if code !== nothing
             ir = codeinst_to_ir(interp, code)
             if isa(ir, IRCode)
-                irsv = IRInterpretationState(interp, ir, mi, sv.world, arginfo.argtypes)
-                rt, nothrow = ir_abstract_constant_propagation(interp, irsv)
+                irinterp = isa(interp, NativeInterpreter) ? NativeInterpreter(interp; irinterp=true) : interp
```

Comment:
Can we pull this out into a generic function that could be overriden?

Reply:
Sure, it is more reasonable.